community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1765,21 +1777,28 @@ def _get_reservation(self, project: str, zone: str, name: str) -> object:
         """See https://cloud.google.com/compute/docs/reference/rest/v1/reservations"""
         return self.compute.reservations().get(
             project=project, zone=zone, reservation=name).execute()
+    
+    @lru_cache()
+    def _get_future_reservation(self, project:str, zone:str, name: str) -> object:
+        """See https://cloud.google.com/compute/docs/reference/rest/v1/futureReservations"""
+        return self.compute.futureReservations().get(project=project, zone=zone, futureReservation=name).execute()
+    
+    
+    def nodeset_reservation(self, reservation_name:str, zone_policy_allow: Optional[list]) -> Optional[ReservationDetails]:
```

Comment:
There is nothing left about `nodeset` in this function. I would recommend to keep it as it was:
```py
def nodeset_reservation(self, nodeset: object) -> Optional[ReservationDetails]:
```

And add separate internal `def _get_reservation(name: str, zone)`

Reply:
Sounds good, just used the existing `_get_reservation` and created the reservation instance in line